src/routes/safe/components/Transactions/TxsTable/columns.tsx

```diff
@@ -105,15 +131,22 @@ const getTransactionTableData = (tx: Transaction, cancelTx: Transaction): TableD
 }
 
 export const getTxTableData = (
-  transactions: List<Transaction>,
+  transactions: List<Transaction | SafeModuleTransaction>,
   cancelTxs: CancellationTransactions,
 ): List<TableData> => {
   return transactions.map((tx) => {
-    if (INCOMING_TX_TYPES[tx.type] !== undefined) {
-      return getIncomingTxTableData(tx)
+    const isModuleTx = [TransactionTypes.SPENDING_LIMIT, TransactionTypes.MODULE].includes(tx.type)
+    const isIncomingTx = INCOMING_TX_TYPES[tx.type] !== undefined
+
+    if (isModuleTx) {
+      return getModuleTxTableData(tx as SafeModuleTransaction)
+    }
+
+    if (isIncomingTx) {
+      return getIncomingTxTableData(tx as Transaction)
     }
 
-    return getTransactionTableData(tx, cancelTxs.get(`${tx.nonce}`))
+    return getTransactionTableData(tx as Transaction, cancelTxs.get(`${tx.nonce}`))
```

Comment:
what was the error there that required `as`?

Reply:
`getTransactionTableData` has issues with tx being of `SafeModuleTransaction` type.

I tried helping TS by using type guards, but it didn't work either: https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types

Reply:
Yeah we should explore using type guards, this is getting a problem for tokens also